File: jelly-vs-tower/typescript/src/Tower.ts
import { ColorType } from './ColorType.js';
import { InvalidLevelException } from './InvalidLevelException.js';

export class Tower {
  readonly id: string;
  readonly color: ColorType;
  readonly level: number;

  constructor(id: string, color: ColorType, level: number) {
    if (level < 1 || level > 4) throw new InvalidLevelException(level);
    this.id = id;
    this.color = color;
    this.level = level;
  }
}

File: jelly-vs-tower/csharp/src/JellyVsTower/Tower.cs
namespace JellyVsTower;

public class Tower
{
    public string Id { get; }
    public ColorType Color { get; }
    public int Level { get; }

    public Tower(string id, ColorType color, int level)
    {
        if (level < 1 || level > 4) throw new InvalidLevelException(level);
        Id = id;
        Color = color;
        Level = level;
    }
}
